event-sources/event-source-file/src/main/java/io/telicent/smart/cache/sources/offsets/file/AbstractJacksonOffsetStore.java

```diff
@@ -71,6 +71,14 @@ public AbstractJacksonOffsetStore(ObjectMapper mapper, File offsetsFile) {
         } else {
             LOGGER.debug("Offsets file {} does not yet exist/is empty, no persistent offsets were loaded as a result",
                          this.offsetsFile.getAbsolutePath());
+            try {
+                this.offsetsFile.createNewFile();
+            } catch (IOException e) {
+                LOGGER.warn("Failed to create offsets file {}: {}", this.offsetsFile.getName(), e.getMessage());
+                throw new IllegalStateException(
+                        String.format("Failed to create offsets file %s", this.offsetsFile.getAbsolutePath()),
+                        e);
+            }
```

Comment:
Good call on testing ability to write to the file location up front, saves things failing later after actual work has happened